templates/registration/login.html

```diff
@@ -0,0 +1,14 @@
+{% extends "base.html" %}
+
+{% block content %}
+  <h1>Login</h1>
+  {% if form.error %}
+    <p>Invalid credentials</p>
+  {% endif %}
+  <form action="{% url "login" %}" method="post">
+    {% csrf_token %}
+    {{ form.as_p }}
+    <input type="submit" value="Submit">
+    <input type="hidden" name="next" value="{{ next }}">

```

Comment:
The `next` hidden input is used to redirect users after login. Ensure that the `next` context variable is correctly passed to the template when rendering the login form.